### ISSUE TITLE
About the output result file

### ISSUE THREAD

**Issue by lbwfff** [Nesvilab/FragPipe#633]

**Comment by fcyu** [comment]
Hi LeeLee,

Please find the description of the result files from https://fragpipe.nesvilab.org/docs/tutorial_fragpipe_outputs.html

In FragPipe, we use the razor protein + mapped protein concept. A peptide is assigned to a razor protein (leading proteins). Other proteins mapped to the peptide are called mapped protein.

Best,

Fengchao

**Comment by lbwfff** [comment]
> Hi LeeLee,
> 
> Please find the description of the result files from https://fragpipe.nesvilab.org/docs/tutorial_fragpipe_outputs.html
> 
> In FragPipe, we use the razor protein + mapped protein concept. A peptide is assigned to a razor protein (leading proteins). Other proteins mapped to the peptide are called mapped protein.
> 
> Best,
> 
> Fengchao

Hi, Fengchao

I still have some confusion, for example, I have the following two items in my index.:
```
>NONHSAG031147.2_3200092_3203369
VAQPGPLEPEEPRAGGRPRRRRDLGSRLQAQRRAQRVAWAEADENEEEAVILAQEEEGVEKPAETHLSGKIGAKKLRKLE
```
```
>sp|Q96HY6|DDRGK_HUMAN DDRGK domain-containing protein 1 OS=Homo sapiens OX=9606 GN=DDRGK1 PE=1 SV=2
....GQEPLHNEELAGAGRVAQPGPLEPEEPRAGGRPRRRRDLGSRLQAQR....
```
There is such a peptide in the combined_peptide.tsv file: `VAQPGPLEPEEPR`

From what I understand, One of the two should be considered as Protein and the other as Mapped Proteins? Because the DDRGK protein also conforms to the enzyme cleavage rules. However, the combined_peptide.tsv file only recognizes NONHSAG031147.2_3200092_3203369 but no DDRGK protein. This confuses me a bit, or I misunderstood something?

Thanks，

LeeLee

**Comment by lbwfff** [comment]
> @lbwfff Do you see both proteins in the PSM or protein tables?

Yes, I looked both tables, both proteins are present, there are some peptides that are thought to be fragments of DDRGK, but not include VAQPGPLEPEEPR.

**Comment by prvst** [comment]
I think it might as indistinguishable (like the decoys), that is why I believe in the confusion, but again, I would have to look at everything to understand what was done.

**Comment by lbwfff** [comment]
> I would have to take a look into your files, but I have the impression you're confusing the assigned protein with the indistinguishable proteins. From the peptide perspective, the program maps every observation to a protein, and the razor algorithm adjusts the assignment based on different criteria. The protein reports will have only the proteins that passed the quality filters and score thresholds, so you might be missing something that got filtered out at the protein level.

Are you talking about the output file? Considering the time difference between us, I will show you tomorrow.

**Comment by lbwfff** [comment]
In my project, I was more concerned with non-uniport proteins, so I noticed proteins like NONHSAG031147.2_3200092_3203369. So I looked up the combined_peptide.tsv file and wondered what evidence supported the existence of NONHSAG031147.2_3200092_3203369, so I found the VAQPGPLEPEEPR fragment.

**Comment by lbwfff** [comment]
Hi, Felipe
I uploaded several result files, of which psm only intercepted part of it because of the file size.

[protein.csv](https://github.com/Nesvilab/FragPipe/files/8415074/protein.csv)
[peptide.csv](https://github.com/Nesvilab/FragPipe/files/8415075/peptide.csv)
[psm.csv](https://github.com/Nesvilab/FragPipe/files/8415110/psm.csv)

**Comment by lbwfff** [comment]
Hi, Felipe
I have another question about Philosopher, I am using fragpipe on a Linux server to analyze Lable-free proteomic data. I find that I spend a lot of time on the ProteinProphet step (maybe over a thousand minutes when I'm analyzing eight samples of data at the same time), one reason is because my index file is very large, another reason may be Philosopher is only using one thread, I would like to ask is there any way to improve this situation?
Thanks,
LeeLee

**Comment by lbwfff** [comment]
Yes, I can't attach here , or I can email you ,what files are needed besides these files and i can also send to you.

**Comment by prvst** [comment]
Please upload all files, with the exception of the raw/mzML to this link https://www.dropbox.com/request/93s56W9LXpOY5cfDehh1

**Comment by lbwfff** [comment]
Hi, Felipe, I have uploaded the file, please take a look

**Comment by prvst** [comment]
@lbwfff I see both proteins in all your files. Protein `Q96HY6|DDRGK_HUMAN` seems to be more abundant than `NONHSAG031147.2_3200092_320336`.

**Comment by lbwfff** [comment]
> @lbwfff I see both proteins in all your files. Protein `Q96HY6|DDRGK_HUMAN` seems to be more abundant than `NONHSAG031147.2_3200092_320336`.

Yes, but aren't we talking about why the software thinks VAQPGPLEPEEPR comes from NONHSAG031147.2_3200092_320336 instead of Q96HY6|DDRGK_HUMAN?

**Comment by prvst** [comment]
We can get some clues from the prot.xml file. I don't see the file in your upload, but there can be different reasons for having the peptide assigned that way. The razor classification is one of them. You can try running the analysis again without the razor assignment to observe if there's any difference in the mapping.

**Comment by guoci** [comment]
@lbwfff The ProteinProphet step should now be much faster if you use the latest release of `Philosopher`
https://github.com/Nesvilab/philosopher/releases/tag/v4.2.1
> Hi, Felipe I have another question about Philosopher, I am using fragpipe on a Linux server to analyze Lable-free proteomic data. I find that I spend a lot of time on the ProteinProphet step (maybe over a thousand minutes when I'm analyzing eight samples of data at the same time), one reason is because my index file is very large, another reason may be Philosopher is only using one thread, I would like to ask is there any way to improve this situation? Thanks, LeeLee

**Comment by lbwfff** [comment]
Hi, Felipe, I have uploaded the prot.xml file, please take a look

**Comment by lbwfff** [comment]
> @lbwfff The ProteinProphet step should now be much faster if you use the latest release of `Philosopher` https://github.com/Nesvilab/philosopher/releases/tag/v4.2.1
> 
> > Hi, Felipe I have another question about Philosopher, I am using fragpipe on a Linux server to analyze Lable-free proteomic data. I find that I spend a lot of time on the ProteinProphet step (maybe over a thousand minutes when I'm analyzing eight samples of data at the same time), one reason is because my index file is very large, another reason may be Philosopher is only using one thread, I would like to ask is there any way to improve this situation? Thanks, LeeLee

I tried running the software again and it was much faster than before, thanks for the update

**Comment by prvst** [comment]
According to your prot.xml, this is the only group where the peptide VAQPGPLEPEEPR is present:

```
<protein_group group_number="14495" probability="0.9934">
      <protein protein_name="NONHSAG031147.2_3200092_3203369" n_indistinguishable_proteins="1" probability="0.9934" percent_coverage="14.0" unique_stripped_peptides="VAQPGPLEPEEPR" group_sibling_id="a" total_number_peptides="10" total_number_distinct_peptides="10" pct_spectrum_ids="0.000" confidence="0.000">
         <parameter name="prot_length" value="93"/>
         <annotation protein_description="NONHSAG031147.2_3200092_3203369"/>
         <peptide peptide_sequence="VAQPGPLEPEEPR" charge="2" initial_probability="0.9989" nsp_adjusted_probability="0.9934" fpkm_adjusted_probability="0.9934" weight="1.00" group_weight="1.00" is_nondegenerate_evidence="Y" n_enzymatic_termini="2" n_sibling_peptides="0.00" n_sibling_peptides_bin="0" n_instances="10" exp_tot_instances="9.25" is_contributing_evidence="Y" calc_neutral_pep_mass="1417.7200">
         </peptide>
      </protein>
</protein_group>
```

The protein Q96HY6|DDRGK_HUMAN has its own protein group, and when we look at the ehader we see this peptide list:

```
<protein protein_name="sp|Q96HY6|DDRGK_HUMAN" n_indistinguishable_proteins="1" probability="1.0000" percent_coverage="47.1" unique_stripped_peptides="AASAGQEPLHNEELAGAGR+AQREAEEAEREER+EHEEYLK+FIYITPEELAAVANFIR+GRAASAGQEPLHNEELAGAGRVAQPGPLEPEEPR+IQDLLAEGTITGVIDDR+IQDLLAEGTITGVIDDRGK+KAREEQAQR+LRLEEEQKEEEER+QRGRVSIAELAQASNSLIAWGR+VSIAELAQASNSLIAWGR+VVLLEDLASQVGLR" group_sibling_id="a" total_number_peptides="94" total_number_distinct_peptides="94" pct_spectrum_ids="0.004" confidence="0.002">
```

Notice how the peptide VAQPGPLEPEEPR is not on the list. We do, however, have GRAASAGQEPLHNEELAGAGR**VAQPGPLEPEEPR**.

This means that the peptide maps uniquely to this protein, and does not map to Q96HY6|DDRGK_HUMAN. That's why you dont see it as an alternative protein.

Hope to have clarified it for you. Please let me know if you have other questions.

**Comment by prvst** [comment]
I'm not sure either, I found the peptide VAQPGPLEPEEPR in only one pep.xml file, so there's a single PSM hit:

```
<spectrum_query start_scan="22843" uncalibrated_precursor_neutral_mass="1417.7151" assumed_charge="2" spectrum="CNHPP_HCC_LC_profiling_L106_T_F1.22843.22843.2" end_scan="22843" index="19964" precursor_neutral_mass="1417.7205" retention_time_sec="1668.6013412475586">
<search_result>
<search_hit peptide="VAQPGPLEPEEPR" massdiff="4.8828125E-4" calc_neutral_pep_mass="1417.72" peptide_next_aa="A" num_missed_cleavages="0" num_tol_term="2" protein_descr="" num_tot_proteins="1" tot_num_ions="24" hit_rank="1" num_matched_ions="9" protein="NONHSAG031147.2_3200092_3203369" peptide_prev_aa="-" is_rejected="0">
<search_score name="hyperscore" value="18.66"/>
<search_score name="nextscore" value="18.66"/>
<search_score name="expect" value="2.033507e-01"/>
</search_hit>
</search_result>
</spectrum_query>
```

What makes me worry is that the peptide GRAASAGQEPLHNEELAGAGRVAQPGPLEPEEPR cannot be found in any of your pep.xml files. Are you sure you're not mixing up results? Did you upload all the files you used?

**Comment by prvst** [comment]
It looks like you did not upload all your files. The list inside prot.xml index is much longer than the list in the Dropbox folder, so I'll just assume the peptide exists in one of those files.

**Comment by fcyu** [comment]
I have the following questions:
1. @lbwfff Are you using PeptideProphet or Percolator?
2. @lbwfff Have you enabled "report alternative protein" in MSFragger tab of FragPipe?
3. @prvst Can you confirm that there is no alternative protein of peptide `VAQPGPLEPEEPR` in the interact.pep.xml?

Thanks,

Fengchao

**Comment by prvst** [comment]
@fcyu see above, I just posted right before you. The pep.xml files I got seems to be just a fraction of all pep.xml used to build the inference, so I assume the peptide must be somewhere out there.

**Comment by lbwfff** [comment]
> I have the following questions:
> 
> 1. @lbwfff Are you using PeptideProphet or Percolator?
> 2. @lbwfff Have you enabled "report alternative protein" in MSFragger tab of FragPipe?
> 3. @prvst Can you confirm that there is no alternative protein of peptide `VAQPGPLEPEEPR` in the interact.pep.xml?
> 
> Thanks,
> 
> Fengchao

I used the default parameter of label-free, which is Percolator, I don't remember if it "report alternative protein", I will check when I get back to the office

**Comment by lbwfff** [comment]
> It looks like you did not upload all your files. The list inside prot.xml index is much longer than the list in the Dropbox folder, so I'll just assume the peptide exists in one of those files.

I ran twelve sets of samples at a time, and given the size of the file, I only uploaded the data for one of the samples that appeared with NONHSAG031147.2_3200092_320336 intensity data, I don't know if this will affect, I will upload other files if needed

**Comment by fcyu** [comment]
Hi @lbwfff ,

It seems to be a bug in MSFragger. Could you please upload the mzML  file of `CNHPP_HCC_LC_profiling_L106_T_F1` to the Dropbox folder? Could you please also upload the `fragger.params` file?

Thanks,

Fengchao

**Comment by lbwfff** [comment]
> fragger.params

Sure, I will upload the .raw format file and fragger.params

**Comment by fcyu** [comment]
> Is the bug affecting the in-silicon enzymatic digestion?

I have no idea before debugging the program.

Best,

Fengchao

**Comment by lbwfff** [comment]
> 

Hi, Fengchao
I downloaded the data from iprox, you can download the data directly from the following address `https://download.iprox.cn/IPX0000937000/IPX0000937001/CNHPP_HCC_LC_profiling_L060_T_F1.raw` 
And below is my fragger.params file.
[fragger.params.zip](https://github.com/Nesvilab/FragPipe/files/8448562/fragger.params.zip)

**Comment by fcyu** [comment]
Hi @lbwfff ,

Thanks for your files. Guo Ci and I fixed the bug in MSFragger. If you want a pre-release version to test, please feel free to let me know.

Best,

Fengchao

**Comment by lbwfff** [comment]
> pre-release version

Hi, Fengchao

Thanks for the fix, I'm more than happy to test the pre-release version and look forward to the final version.

Best,

LeeLee

**Comment by fcyu** [comment]
Sounds good! Could you send an email to me: yufe@umich.edu

Thanks,

Fengchao

**Comment by lbwfff** [comment]
> Sounds good! Could you send an email to me: [yufe@umich.edu](mailto:yufe@umich.edu)
> 
> Thanks,
> 
> Fengchao

I sent you the email already, please check mailbox.

**Comment by fcyu** [comment]
Sent you the link to download.

Best,

Fengchao
